tests/io/test.xpi.js

```diff
@@ -230,20 +287,61 @@ describe('Xpi.getFileAsStream()', function() {
 
   });
 
-  it('should reject if path does not exist', () => {
+  it('should reject if error in openReadStream', () => {
     var myXpi = new Xpi('foo/bar', this.fakeZipLib);
     myXpi.files = {
       'install.rdf': installRdfEntry,
-      'chrome.manifest': chromeManifestEntry,
     };
 
-    return myXpi.getFileAsStream('whatever')
+    this.openStub.yieldsAsync(null, this.fakeZipFile);
+    this.openReadStreamStub.yieldsAsync(
+      new Error('getChunkAsBuffer openReadStream test'));
+
+    return myXpi.getChunkAsBuffer('install.rdf')
       .then(unexpectedSuccess)
       .catch((err) => {
-        assert.include(err.message, 'Path "whatever" does not exist');
+        assert.include(err.message, 'getChunkAsBuffer openReadStream test');
+      });
+  });
+
+  it('should resolve with a buffer', () => {
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+    myXpi.files = {
+      'install.rdf': installRdfEntry,
+    };
+
+    this.openStub.yieldsAsync(null, this.fakeZipFile);
+
+    var rstream = new Readable();
+    rstream.push('123\n');
+    rstream.push(null);
+
+    this.openReadStreamStub.yields(null, rstream);
+
+    return myXpi.getChunkAsBuffer('install.rdf', 2)
+      .then((buffer) => {
+        assert.equal(buffer.toString(), '12');
       });
   });
 
+});
+
+
+describe('Xpi.getFileAsStream()', function() {
+
+  beforeEach(() => {
```

Comment:
If these were copy + pasted we could turn this into a helper and `import` it rather than redeclaring it, but this amount of copy/pasting doesn't bother me.